plugins/woocommerce/client/blocks/assets/js/blocks/product-gallery/inner-blocks/product-gallery-next-previous-buttons/edit.tsx

```diff
@@ -2,19 +2,60 @@
  * External dependencies
  */
 import { useBlockProps } from '@wordpress/block-editor';
+import clsx from 'clsx';
+import type { BlockAttributes } from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
 import { PrevIcon, NextIcon } from './icons';
 
-export const Edit = (): JSX.Element => {
+const getVerticalAlignmentClass = ( verticalAlignment?: string ) => {
+	if ( verticalAlignment === 'top' ) {
+		return 'aligntop';
+	}
+	if ( verticalAlignment === 'bottom' ) {
+		return 'alignbottom';
+	}
+	// Default to center.
+	return '';
+};
+
+const getAlignClass = ( align?: string ) => {
+	if ( align === 'left' ) {
+		return 'alignleft';
+	}
+	if ( align === 'right' ) {
+		return 'alignright';
+	}
+	if ( align === 'center' ) {
+		return 'aligncenter';
+	}
+	// Default to spread.
+	return '';
+};
+
+export const Edit = ( { attributes }: { attributes: BlockAttributes } ) => {
 	const blockProps = useBlockProps( {
 		className: 'wc-block-product-gallery-large-image-next-previous__button',
 	} );
 
+	// We're passing blockProps to buttons because they're the main block content.
+	// However, alignment has to be applied to the container hence we're handling
+	// these attributes "manually" here.
```

Comment:
Is there a specific reason why we don't pass `blockProps` to the container?

Reply:
Good point! Based on our sync elsewhere I changed the approach and provide `blockProps` to container but provide the specific local styles classes and style to the buttons.

Right now:
- global styles work on both editor and frontend
- local styles override global styles on both editor and frontend

The only issue I found is global styles margin are not working on the frontend as they're overwritten by core layout styles due to higher specificity (it's a GB bug: https://github.com/WordPress/gutenberg/issues/44736)

![Screen Shot 2025-04-24 at 12 34 44 PM](https://github.com/user-attachments/assets/49f616d7-5308-4bbb-8e25-081b5699fd7b)

I'd appreciate one more review! 🙇